### ISSUE TITLE
Error with influxdb_user.present grants

### ISSUE THREAD

**Issue by kris-anderson** [saltstack/salt#44773]
### Description of Issue/Question
I'm trying to create a user for influxdb and granting them access to specific databases. However, when I run `state.apply` I get the following error:
```
          ID: telegraf user present in influxdb
    Function: influxdb_user.present
        Name: telegraf
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/local/lib/python2.7/site-packages/salt/state.py", line 1843, in call
                  **cdata['kwargs'])
                File "/usr/local/lib/python2.7/site-packages/salt/loader.py", line 1795, in wrapper
                  return f(*args, **kwargs)
                File "/usr/local/lib/python2.7/site-packages/salt/states/influxdb_user.py", line 86, in present
                  for database, privilege in grants.items():
              AttributeError: 'list' object has no attribute 'items'
     Started: 15:38:48.555118
    Duration: 6.701 ms
     Changes:
```

### Setup
Here is the state I'm trying to run:
```
telegraf user present in influxdb:
  influxdb_user.present:
    - name: telegraf
    - password: supersecret
    - admin: False
    - grants:
      - telegraf: all
```

If I remove this section the state runs successfully.
```
    - grants:
      - telegraf: all
```

This however, says it should be supported?
https://docs.saltstack.com/en/latest/ref/states/all/salt.states.influxdb_user.html#management-of-influxdb-users

### Versions Report
```
Salt Version:
           Salt: 2017.7.2

Dependency Versions:
           cffi: 1.7.0
       cherrypy: Not Installed
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.9.5
        libgit2: 0.25.1
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: 2.10
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.25.1
         Python: 2.7.14 (default, Nov  4 2017, 01:15:42)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.2
            ZMQ: 4.2.2

System Versions:
           dist:
         locale: US-ASCII
        machine: amd64
        release: 11.1-RELEASE-p4
         system: FreeBSD
        version: Not Installed
```

**Comment by kris-anderson** [comment]
Looks like I got the structure wrong. In case someone happens to find this through a search, this seems to have worked:

```
    - grants:
      telegraf: all
```

**Comment by kris-anderson** [comment]
Scratch that... the error goes away but it doesn't actually apply the grant.

**Comment by garethgreenaway** [comment]
@kris-anderson Thanks for the report.  Can you double check the formatting in your state file and ensure there are spaces in front of the telegraf grant?  Something like this:
```
- grants:
    telegraf: all
```

**Comment by kris-anderson** [comment]
That works! Thank you so much for the help. I created a PR to update the documentation with an example of the formatting in case it helps someone else in the future:

https://github.com/saltstack/salt/pull/44788

**Comment by rallytime** [comment]
Closed via #44788
